lib/jxl/dec_tone_mapping-inl.h

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#if defined(LIB_JXL_DEC_TONE_MAPPING_INL_H_) == defined(HWY_TARGET_TOGGLE)
+#ifdef LIB_JXL_DEC_TONE_MAPPING_INL_H_
+#undef LIB_JXL_DEC_TONE_MAPPING_INL_H_
+#else
+#define LIB_JXL_DEC_TONE_MAPPING_INL_H_
+#endif
+
+#include <hwy/highway.h>
+
+#include "lib/jxl/transfer_functions-inl.h"
+
+HWY_BEFORE_NAMESPACE();
+namespace jxl {
+namespace HWY_NAMESPACE {
+namespace {
+
+template <typename D, typename V>
+class Rec2408ToneMapper {
+ public:
+  explicit Rec2408ToneMapper(D df, std::pair<float, float> source_range,
```

Comment:
Is there a reason to pass a df instead of constructing one where needed?

Reply:
(one reason could be, if this was not a constructor but a free function, to get deduction for the template parameter)

Reply:
Good point, updated.